web-services-app/src/main/java/edu/unc/lib/boxc/web/services/rest/modify/ThumbnailController.java

```diff
@@ -123,36 +134,49 @@ public ResponseEntity<Object> assignThumbnail(@PathVariable("pidString") String
             log.error("Error assigning file {} as thumbnail", request.getFilePidString(), e);
             return new ResponseEntity<>(HttpStatus.INTERNAL_SERVER_ERROR);
         }
-        return new ResponseEntity<>(HttpStatus.OK);
+        result.put("timestamp", System.currentTimeMillis());
+        result.put("newThumbnailId", object.getPid().getId());
+        return new ResponseEntity<>(result, HttpStatus.OK);
     }
 
     @DeleteMapping(value = "/edit/deleteThumbnail/{pidString}")
     @ResponseBody
     public ResponseEntity<Object> deleteThumbnail(@PathVariable("pidString") String pidString) {
         PID pid = PIDs.get(pidString);
+        Map<String, Object> result = new HashMap<>();
+        result.put(ACTION, "deleteThumbnail");
 
         AccessGroupSet principals = getAgentPrincipals().getPrincipals();
         aclService.assertHasAccess("Insufficient permissions to assign thumbnail for " + pidString,
                 pid, principals, Permission.editDescription);
 
         var object = repositoryObjectLoader.getRepositoryObject(pid);
-        if (!(object instanceof FileObject)) {
-            log.error("Error object is not a file: {}", pidString);
+        if (!(object instanceof FileObject) && !(object instanceof WorkObject)) {
+            log.error("Error object is not a file or work: {}", pidString);
             return new ResponseEntity<>(HttpStatus.BAD_REQUEST);
         }
 
         var agent = AgentPrincipalsImpl.createFromThread();
         var request = new ThumbnailRequest();
         request.setAgent(agent);
-        request.setFilePidString(pidString);
         request.setAction(ThumbnailRequest.DELETE);
+        if (object instanceof WorkObject) {
+            var fileId = ((WorkObject) object).getThumbnailObject().getPid().getId();
```

Comment:
what happens if the user manages to call deleteThumbnail on a work that doesn't have a thumbnail assigned? I'm guessing a NullPointerException. Might be nicer to respond with a BAD_REQUEST or similar